src/lib/controllers/texts-controller.js

```diff
@@ -0,0 +1,82 @@
+// import { v4 as uuidv4 } from 'uuid'
+
+import Alignment from '@/lib/data/alignment'
+import DownloadController from '@/lib/controllers/download-controller.js'
+import UploadController from '@/lib/controllers/upload-controller.js'
+
+export default class TextsController {
+  createAlignment (originDocSource) {
+    this.alignment = new Alignment(originDocSource)
+  }
+
+  updateOriginDocSource (originDocSource) {
+    if (!this.alignment) {
+      this.createAlignment(originDocSource)
+    } else {
+      this.alignment.updateOriginDocSource(originDocSource)
+    }
+  }
+
+  updateTargetDocSource (targetDocSource) {
+    if (!this.alignment) {
+      this.createAlignment()

```

Comment:
no, we don't. We start to create an alignment from getting selected originDocSource first.
And in this case we should throw an error or create incorrect alignment without originDocSource.

But now I think it would be better to throw an error. Will fix.

Reply:
fixed